### PR TITLE
Feature/zipkin support redux

### PR DESCRIPTION
**Pull request by lloydmeta**
Redo of #110 to move Zipkin tracing further down the stack.

### CODE REVIEW

File: app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala [hunk 1]
```diff
@@ -80,7 +89,11 @@ trait HttpPartRequestHandler extends Handler {
       val maybeBody = hArgs.collectFirst {
         case (p, values) if p.paramType == ParamType.Body && values.nonEmpty => values.head
       }
-      val headers = collectHeaders(hArgs) ++ buildTracingHeaders(partRequestInfo)
+      val headers = {
+        collectHeaders(hArgs) ++
+          buildTracingHeaders(partRequestInfo) ++
+          tracingSpan.fold(Map.empty[String, String])(zipkinService.spanToIdsMap)
```
Comment (mauhiz) [inline app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala]: Is there a test to make sure the headers are passed?

File: app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala [hunk 2]
```diff
@@ -52,27 +57,31 @@ trait HttpPartRequestHandler extends Handler {
...
-        fromFallback = true,
-        message = "From fallback"
-      )
-    ).map {
-        createPartResponse
-      }
+  def process(partRequestInfo: PartRequestInfo, hArgs: HandlerArguments)(implicit parentSpan: Span): Future[PartResponse] = {
+    TracedFuture(s"Http request for - ${partRequestInfo.partRequest.partId}") { maybeSpan =>
+      hystrixExecutor.future(
+        createBlockingHttpRetrieve(partRequestInfo, hArgs, maybeSpan).retrieve().copy(),
```
Comment (mauhiz) [inline app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala]: Please remove that useless `copy()` (I know it is not yours)

Comment (lloydmeta) [inline app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala]: Good point, this was strange. Removed.

File: app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala (hunk 1 above)
Comment (lloydmeta) [inline app/com/m3/octoparts/aggregator/handler/HttpPartRequestHandler.scala]: Now there is !

**Comment by mauhiz** [comment]
LGTM

**Pull request merged**